explorer/src/components/JupiterSwapModal.tsx

```diff
@@ -0,0 +1,562 @@
+	const [tokenSearch, setTokenSearch] = useState("");
+
+	const [formValue, setFormValue] = useState<UseJupiterProps>({
+		amount: 1 * 10 ** 6,
+		inputMint: inputMint ? new PublicKey(inputMint) : undefined,
+		outputMint: outputMint ? new PublicKey(outputMint) : undefined,
+		slippage: 0.1,
+	});
+	const [inputBalance, setInputBalance] = useState(0);
+
+	const [showSlippageSettings, setShowSlippageSettings] =
+		useState<boolean>(false);
+	const [selectSlippage, setSelectSlippage] = useState<number | null>(null);
+	const [inputSlippage, setInputSlippage] = useState("");
+
+	const [showAggregatorSearch, setShowAggregatorSearch] =
+		useState<boolean>(false);
+	const [aggregator, setAggregator] = useState<Aggregator | null>(null);
+	const [aggregatorOptions, setAggregatorOptions] = useState<Array<Aggregator>>(aggregators);
+	const [aggregatorSearch, setAggregatorSearch] = useState("");
+
+	const [showMore, setShowMore] = useState(false);
+
+	const setWalletBalance = async (tokenAddress: string, type: string) => {
+		if (wallet?.publicKey) {
+			try {
+				const info = await connection.getParsedTokenAccountsByOwner(wallet.publicKey, {
+					mint: new PublicKey(tokenAddress),
+				},
+				);
+				const balance = info.value[0].account.data.parsed.info.tokenAmount.uiAmount;
+				if (type === "pay") {
+					setBalance(balance);
+					if ((balance) - 0.00001 > 0) {
+						setInputBalance(balance - 0.00001);
+					} else {
+						setInputBalance(0);
+					}
+				} else if (type === "receive") {
+					setRecieveTokenBalance(balance);
+				}
+			} catch (error) {
+				console.log(error);
+			}
+		}
+	}
+
+	useEffect(() => {
+		setTokenLoading(true);
+		fetch(TOKEN_LIST_URL["mainnet-beta"])
+			.then((response) => {
+				response.json().then((tokens) => {
+					setAllTokens(tokens);
+					setTokens(tokens);
+				});
+			}).catch((error) => {
+				console.error(error);
+			}).finally(() => {
+				setTokenLoading(false);
+			});
+	}, []);
+
+
+	const [inputTokenInfo, outputTokenInfo] = useMemo(() => {
+		return [
+			tokenMap.get(formValue.inputMint?.toBase58() || ""),
+			tokenMap.get(formValue.outputMint?.toBase58() || ""),
+		];
+	}, [formValue.inputMint?.toBase58(), formValue.outputMint?.toBase58()]);
+
+	const amountInDecimal = useMemo(() => {
+		return formValue.amount * 10 ** (inputTokenInfo?.decimals || 1);
+	}, [inputTokenInfo, formValue.amount]);
+
+	const setTokenValues = async (token: any, type?: string) => {
+		if (tokenSearchType === 'pay' || type === 'pay') {
+			setWalletBalance(token.address, "pay");
+			setFormValue((prevValue) => {
+				return {
+					...prevValue,
+					inputMint: new PublicKey(token.address),
+				};
+			});
+
+			let publicKey = wallet.publicKey || new PublicKey(token.address);
+			const info = await connection.getParsedTokenAccountsByOwner(publicKey, {
+				mint: new PublicKey(token.address),
+			});
+			const balance = info.value[0].account.data.parsed.info.tokenAmount.uiAmount;
```

Comment:
<img width="963" alt="image" src="https://user-images.githubusercontent.com/20179273/185062756-9ddf5ad3-d58a-47d7-a5a5-9bb014ad65a6.png">

Reply:
Seems to have an issue